extensions/skyportal/static/js/components/VegaPlotZTFArchive.jsx

```diff
@@ -0,0 +1,191 @@
+            "join([format(datum.mag, '.2f'), ' ± ', format(datum.magerr, '.2f')], '')",
+          as: "magAndErr",
+        },
+        { calculate: `${jdNow} - datum.hjd`, as: "daysAgo" },
+      ],
+      encoding: {
+        x: {
+          field: "daysAgo",
+          type: "quantitative",
+          scale: {
+            zero: false,
+            reverse: true,
+          },
+          axis: {
+            title: "days ago",
+          },
+        },
+        y: {
+          field: "mag",
+          type: "quantitative",
+          scale: {
+            zero: false,
+            reverse: true,
+          },
+          axis: {
+            title: "mag",
+          },
+        },
+        color: {
+          field: "filter",
+          type: "nominal",
+          scale: color_scale,
+        },
+        tooltip: [
+          { field: "magAndErr", title: "mag", type: "nominal" },
+          { field: "filter", type: "ordinal" },
+          { field: "hjd", type: "quantitative" },
+          { field: "daysAgo", type: "quantitative" },
+          { field: "catflags", type: "ordinal" },
+          { field: "programid", type: "ordinal" },
+          { field: "chi", type: "quantitative" },
+          { field: "sharp", type: "quantitative" },
+        ],
+        opacity: {
+          condition: { selection: "filterMags", value: 1 },
+          value: 0,
+        },
+      },
+    },
+
+    // Render error bars
+    {
+      selection: {
+        filterErrBars: {
+          type: "multi",
+          fields: ["filter"],
+          bind: "legend",
+        },
+      },
+      transform: [
+        { filter: "datum.mag != null && datum.magerr != null" },
+        { calculate: "datum.mag - datum.magerr", as: "magMin" },
+        { calculate: "datum.mag + datum.magerr", as: "magMax" },
+        { calculate: `${jdNow} - datum.hjd`, as: "daysAgo" },
+      ],
+      mark: {
+        type: "rule",
+        size: 0.5,
+      },
+      encoding: {
+        x: {
+          field: "daysAgo",
+          type: "quantitative",
+          scale: {
+            zero: false,
+            reverse: true,
+          },
+          axis: {
+            title: "days ago",
+          },
+        },
+        y: {
+          field: "magMin",
+          type: "quantitative",
+          scale: {
+            zero: false,
+            reverse: true,
+          },
+        },
+        y2: {
+          field: "magMax",
+          type: "quantitative",
+          scale: {
+            zero: false,
+            reverse: true,
+          },
+        },
+        color: {
+          field: "filter",
+          type: "nominal",
+          scale: color_scale,
+          legend: {
+            orient: isMobileOnly ? "bottom" : "right",
+          },
+        },
+        opacity: {
+          condition: { selection: "filterErrBars", value: 1 },
+          value: 0,
+        },
+      },
+    },
+  ],
+});
+
+const VegaPlot = withOrientationChange(({ data, color_scale, isPortrait }) => (
```

Comment:
Any reason `color_scale` isn't in camel case?

Reply:
No reason, fixed.